src/integrationTest/java/org/openstreetmap/atlas/geography/atlas/raw/OsmPbfToSlicedRawAtlasTest.java

```diff
@@ -0,0 +1,109 @@
+package org.openstreetmap.atlas.geography.atlas.raw;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.pbf.AtlasLoadingOption;
+import org.openstreetmap.atlas.geography.atlas.pbf.OsmPbfLoader;
+import org.openstreetmap.atlas.geography.atlas.raw.creation.RawAtlasGenerator;
+import org.openstreetmap.atlas.geography.atlas.raw.slicing.RawAtlasCountrySlicer;
+import org.openstreetmap.atlas.geography.boundary.CountryBoundaryMap;
+import org.openstreetmap.atlas.locale.IsoCountry;
+import org.openstreetmap.atlas.streaming.compression.Decompressor;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.resource.InputStreamResource;
+import org.openstreetmap.atlas.tags.ISOCountryTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+
+/**
+ * Tests OSM PBF --> Raw Atlas --> Sliced Raw Atlas flow.
+ *
+ * @author mgostintsev
+ */
+public class OsmPbfToSlicedRawAtlasTest
+{
+    private static CountryBoundaryMap COUNTRY_BOUNDARY_MAP;
+    private static Set<IsoCountry> COUNTRIES;
+
+    static
+    {
+        COUNTRIES = new HashSet<>();
+        COUNTRIES.add(IsoCountry.forCountryCode("CIV").get());
+        COUNTRIES.add(IsoCountry.forCountryCode("GIN").get());
+        COUNTRIES.add(IsoCountry.forCountryCode("LBR").get());
+    }
+
+    @BeforeClass
+    public static void setup()
+    {
+        COUNTRY_BOUNDARY_MAP = new CountryBoundaryMap(
+                new InputStreamResource(() -> OsmPbfToSlicedRawAtlasTest.class
+                        .getResourceAsStream("CIV_GIN_LBR_osm_boundaries.txt.gz"))
+                                .withDecompressor(Decompressor.GZIP));
+    }
+
+    @Test
+    @Ignore("Taking too long, but still valuable to have")
```

Comment:
Might be worth making smaller?

Reply:
Yeah, possibly. I have an optimization in mind for assigning country codes for Points - that's the part that's taking the most right now. Instead of assigning each individual Point a country code - only assign ones to existing ones on the boundary or newly created ones. All other ones are guaranteed to be points in lines that are contained within one country. Toying with this idea now!